### DISCUSSION TITLE
How to debug through VS Code debugger?

### DISCUSSION THREAD

**Discussion by alexmurari** [webpack/webpack#16030] [Q&A]
Hi, 

I'm fairly new to Webpack and I was studying Typescript using the code from [this repo](https://github.com/DanWahlin/typescript-fundamentals). They use Webpack 5 to build/bundle the app code. It's just plain TypeScript + Webpack 5 with some simple HTML files.

I can run `webpack serve --open` and everything works fine and I can debug from the browser dev tools. But how can I debug using the VS Code debugger? It confuses me because webpack generates the files in memory when launching the server, so attaching the VS Code debugger, pointing to the source map files is kind of tricky and I don't really understand how to put those pieces together.

I was trying to create a `launch.json` file to handle the build and launch process but with no success (maybe `webpack serve` is not the right command to work with the VS Code debugger).

Any ideas?

Thanks.

**Comment by vankop** [comment]
you probably need to specify a path there like `node_modules/.bin/webpack`, not sure

**Reply by alexmurari** [reply]
Thanks for the sugestion, @vankop! I'm gonna try It.

But what does this path means and how it is useful? It only contains the webpack bins, right?

**Reply by vankop** [reply]
it is webpack script path ( in `bin` folder), in `package.json` scripts it is not required to put path `node_modules/.bin/` before script name
